test/jdk/java/util/Locale/Bug8179071.java

```diff
@@ -38,53 +38,60 @@
-                                                          "uz-AF" , "yan",
-                                                          "sr-ME", "\u0458\u0430\u043d",
-                                                          "scc", "\u0458\u0430\u043d",
-                                                          "sh", "jan",
-                                                          "ha-Latn-NE", "Jan",
-                                                          "i-lux", "Jan.");
 
-
-    private static void test(String tag, String expected) {
+    // Ensure the display name for the given tag's January is correct
+    @ParameterizedTest
+    @MethodSource("shortJanuaryNames")
+    public void janDisplayNameTest(String tag, String expected) {
         Locale target = Locale.forLanguageTag(tag);
         Month day = Month.JANUARY;
         TextStyle style = TextStyle.SHORT;
         String actual = day.getDisplayName(style, target);
-        if (!actual.equals(expected)) {
-            throw new RuntimeException("failed for locale  " + tag + " actual output " + actual +"  does not match with  " + expected);
-        }
+        assertEquals(expected, actual);
     }
 
-    /**
-     * getAvailableLocales() should not contain any deprecated or Legacy language tags
-     */
-    private static void checkInvalidTags() {
-        Set<String> invalidTags = new HashSet<>();
-        Arrays.asList(Locale.getAvailableLocales()).stream()
-                .map(loc -> loc.toLanguageTag())
-                .forEach( tag -> {if(LegacyAliases.contains(tag)) {invalidTags.add(tag);}});
-        if (!invalidTags.isEmpty()) {
-          throw new RuntimeException("failed: Deprecated and Legacy tags found  " + invalidTags  + " in AvailableLocales ");
-        }
+    // Expected month format data for locales after language aliases replacement.
+    private static Stream<Arguments> shortJanuaryNames() {
+        return Stream.of(
+                Arguments.of("pa-PK", "\u0a1c\u0a28"),
+                Arguments.of("uz-AF" , "yan"),
+                Arguments.of("sr-ME", "\u0458\u0430\u043d"),
+                Arguments.of("scc", "\u0458\u0430\u043d"),
+                Arguments.of("sh", "jan"),
+                Arguments.of("ha-Latn-NE", "Jan"),
+                Arguments.of("i-lux", "Jan.")
+        );
     }
 
-    public static void main(String[] args) {
-        shortJanuaryNames.forEach(Bug8179071::test);
-        checkInvalidTags();
+    // getAvailableLocales() should not contain any deprecated or Legacy language tags
+    @Test
+    public void invalidTagsTest() {
+        Set<String> invalidTags = new HashSet<>();
+        Arrays.stream(Locale.getAvailableLocales())
+                .map(Locale::toLanguageTag)
+                .forEach(tag -> {if(LegacyAliases.contains(tag)) {invalidTags.add(tag);}});
+        assertEquals(true, invalidTags.isEmpty(),
+                "Deprecated and Legacy tags found  " + invalidTags  + " in AvailableLocales ");
```

Comment:
```suggestion
                "Deprecated and Legacy tags found  " + invalidTags + " in AvailableLocales ");
```

Reply:
Thanks, updated this and the other suggested WS change